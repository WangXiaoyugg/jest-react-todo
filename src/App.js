import React from 'react';

function App() {
  return (
    <div className="container" title="garen" data-test="container">
      Hello World
      <div className="header">Header</div>
    </div>
  );
}

export default App;
