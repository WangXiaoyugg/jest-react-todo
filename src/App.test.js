import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import App from './App';

it('renders without crashing', () => {
  const wrapper = shallow(<App />);
  const container = wrapper.find('[data-test="container"]');
  // expect(container.length).toBe(1);
  // // console.log(wrapper.debug());
  // expect(container.prop('title')).toBe('garen');
  expect(container).toExist();
  expect(container).toHaveProp('title', 'garen');
  expect(wrapper).toMatchSnapshot();
});
