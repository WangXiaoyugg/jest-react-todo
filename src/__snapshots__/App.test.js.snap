// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="container"
  data-test="container"
  title="garen"
>
  Hello World
  <div
    className="header"
  >
    Header
  </div>
</div>
`;
